### ISSUE TITLE
Reference key in Signature

### ISSUE THREAD

**Issue by leinfink** [pprevos/citar-denote#24]
Hi! I was thinking about an option of adding the reference key as the denote signature. This way, one could have the reference in the filename while still retaining the option of using descriptive titles like author-year etc.

Having the reference in the filename might also possibly make the lookup for reference notes easier, I suspect, but I don't know how that is solved currently.

What do you think? I could try do this myself and make a PR.

**Comment by pprevos** [comment]
Hi Henrik,

It is a good idea to make file names more recognisable using citation key as signature, but it can only be used if citation keys are limited to alphanumeric characters.

The main issue with using citation keys in Denote filenames is that it limits how they are constructed. BibTeX allows alphanumeric, dashes, underscores colons and other punctuation marks in citation keys.

For example, when my citation key is `einstein_1905`, then Denote filename will be: `20231121T071320==einstein=1905--testing__bib.org`. The underscore is change to an equals sign. When using `eistein-1905` as citation key, the filename becomes: `20231121T075355==einstein1905-testing__bib.org`. Denote forces the signature to comply with its specification.

Happy to receive any PR with a solution you had in mind.

**Comment by leinfink** [comment]
Hm, I didn't think of that limitation. I don't have a solution for that at the moment. I will just try to make it work for my own use-case then (my bibkeys are alphanumeric), and maybe we can integrate it as an optional setting.

**Comment by pprevos** [comment]
Should not be hard to implement. As long as it does not change the way the program links notes to literature.

I was already working on a version to manually add signatures, so I changed it a bit. Try this patch to see if it does what you need. It still prompts for a signature when creating a note, but hitting enter will use the citation key. Need to find a way to stop the prompting alltogether

```
(defcustom citar-denote-signature 'citekey
  "Ask for a signature or use the citation key as signature.

When using citation keys, Denote removes or converts non-alphanumeric symbols
for comparability with the file-naming convention."
  :group 'citar-denote
  :type '(choice
          (const :tag "User prompt" 'prompt)
          (const :tag "Citation key" 'citekey)))

;;;###autoload
(defun citar-denote--create-note (citekey &optional _entry)
  "Create a bibliographic note for CITEKEY with properties ENTRY.

The note file type is determined by `citar-denote-file-type'.

The title format is set by `citar-denote-title-format'.

When `citar-denote-subdir' is non-nil, prompt for a subdirectory.

`citar-denote-signature' either prompt for a signature or uses CITEKEY."
  (denote
   (read-string "Title: " (citar-denote--generate-title citekey))
   (citar-denote--keywords-prompt citekey)
   citar-denote-file-type
   (when citar-denote-subdir (denote-subdirectory-prompt))
   nil
   (when citar-denote-template (denote-template-prompt))
   (cond
    ((eq citar-denote-signature 'prompt)
     (denote-signature-prompt))
    ((eq citar-denote-signature 'citekey)
     (denote-signature-prompt citekey nil))
    (t nil)))
  (citar-denote--add-reference citekey citar-denote-file-type))
```

**Comment by pprevos** [comment]
Hi Henrik, just published the new version of Citar-Denote ([0c68a4f](https://github.com/pprevos/citar-denote/commit/0c68a4f74f5dcfc23ef03b211658a2ccc5d9aa3b)) that lets you use citation keys as signatures (within the limitations of the Denote file naming convention).

**Comment by leinfink** [comment]
Cool stuff, thanks!
